routes/routes.go

```diff
@@ -51,20 +53,38 @@ type Decorator interface {
 	Decorate(Handler) Handler
 }
 
+// csvGenerator is an interface for any type that can generate a CSV file content
+type csvGenerator interface {
+	ToCSVable() [][]string
+}
+
 func resetRegisteredHandlers() {
 	RegisteredHandlers = RegisteredHandlers[:0]
 }
 
 func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	arguments := make(Arguments)
 	status, output := h.Func(w, r, arguments)
-	w.Header()["Content-Type"] = []string{"application/json; charset=utf-8"}
-	w.WriteHeader(status)
-	e := json.NewEncoder(w)
-	if config.PrettyJsonResponses {
-		e.SetIndent("", "\t")
+	contentType := r.Header["Content-Type"][0]
+	w.Header()["Content-Type"] = []string{fmt.Sprintf("%s; charset=utf-8", contentType)}
+	switch contentType {
+	case "application/json":
```

Comment:
HTTP Content-Type negociation is more complicated than that : user agents can specify a list of accepted content types, with prioritization, and usage of wildcards. For example a client may request ```Accept: text/*;q=0.5 application/json;q=0.8``` where `application/json` should be selected. I think this should be implement as a multiplexer like `routes.MethodMuxer` and as a decorator.